File: content-interface/src/main/java/com/service/AttributeService.java
package com.service;

import com.shopxx.pojo.XxAttribute;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface AttributeService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<XxAttribute> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(XxAttribute attribute);
	
	
	/**
	 * 修改
	 */
	public void update(XxAttribute attribute);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public XxAttribute findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(XxAttribute attribute, int pageNum, int pageSize);
	
}

File: manager-web/src/main/webapp/js/service/attributeService.js
//服务层
app.service('attributeService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../attribute/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../attribute/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../attribute/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../attribute/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../attribute/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../attribute/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../attribute/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
